homeassistant/components/legrand_rflc/config_flow.py

```diff
@@ -0,0 +1,171 @@
+            _LOGGER.warning("OS getaddrinfo %s error %s", self.HOST, error)
+            return self.async_abort(reason=self.ABORT_NO_DEVICES_FOUND)
+        address = discovery_info[IP_ADDRESS]
+        if any(
+            resolution[4][0] == address
+            for resolution in resolutions
+            if resolution[0] == socket.AF_INET
+        ):
+            await self._async_handle_discovery_without_unique_id()
+            # wait for user interaction in the next step
+            return await self.async_step_user()
+        _LOGGER.warning("%s does not resolve to discovered %s", self.HOST, address)
+        return self.async_abort(reason=self.ABORT_NO_DEVICES_FOUND)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        host = self.HOST
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            await self.async_set_unique_id(host)  # already_in_progress?
+            self._abort_if_unique_id_configured()  # already_configured?
+            key = None
+            kwargs = {"key": key, "loop_timeout": -1}
+            if CONF_PASSWORD in user_input:
+                key = kwargs["key"] = lc7001.aio.hash_password(
+                    user_input[CONF_PASSWORD].encode()
+                )
+            if CONF_PORT in user_input:  # for testing server emulation on localhost
+                kwargs["port"] = user_input[CONF_PORT]
+            task = self.hass.async_create_task(
+                lc7001.aio.Connector(host, **kwargs).loop()
+            )
+            try:
+                mac = await task
+            except OSError:
+                errors[CONF_HOST] = self.ERROR_INVALID_HOST
+            except lc7001.aio.Authenticator.Error:
+                errors[CONF_PASSWORD] = self.ERROR_INVALID_AUTH
+            else:
+                data = {CONF_HOST: host, CONF_MAC: mac}
+                if key is not None:
+                    data[CONF_AUTHENTICATION] = key.hex()
+                return self.async_create_entry(title=host, data=data)
+
+        # get user_input
+        return self.async_show_form(
+            step_id="user",
+            data_schema=voluptuous.Schema(
+                {
+                    voluptuous.Required(CONF_HOST, default=host): str,
+                    voluptuous.Optional(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle configuration by reauth."""
+        host = self.context["unique_id"]
+        errors = {CONF_PASSWORD: self.ERROR_INVALID_AUTH}
+        if user_input is not None:
+            key: bytes | None = None
+            kwargs = {"key": key, "loop_timeout": -1}
+            if CONF_AUTHENTICATION in user_input:
```

Comment:
How do you get here, I don't see this in your data schema?

Reply:
CONF_AUTHENTICATION is derived (hashed) from the CONF_PASSWORD from user_input

Reply:
OK, that makes sense, thx for clearing that up for me.